src/core/seqner.rs

```diff
@@ -0,0 +1,268 @@
+use lazy_static::lazy_static;
+
+use crate::core::matter::{tables as matter, Matter};
+use crate::error::{err, Error, Result};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Seqner {
+    raw: Vec<u8>,
+    code: String,
+    size: u32,
+}
+
+impl Default for Seqner {
+    fn default() -> Self {
+        Seqner { raw: vec![], code: matter::Codex::Salt_128.to_string(), size: 0 }
+    }
+}
+
+fn validate_code(code: &str) -> Result<()> {
+    lazy_static! {
+        static ref CODES: Vec<&'static str> = vec![matter::Codex::Salt_128,];
+    }
+
+    if !CODES.contains(&code) {
+        return err!(Error::UnexpectedCode(code.to_string()));
+    }
+
+    Ok(())
+}
+
+impl Seqner {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        sn: Option<u128>,
+        snh: Option<&str>,
+        code: Option<&str>,
+        raw: Option<&[u8]>,
+        qb64b: Option<&mut Vec<u8>>,
+        qb64: Option<&str>,
+        qb2: Option<&mut Vec<u8>>,
+        strip: Option<bool>,
+    ) -> Result<Self> {
+        let buf;
+        let raw: Option<&[u8]> =
+            if raw.is_none() && qb64b.is_none() && qb64.is_none() && qb2.is_none() {
+                let sn = if let Some(sn) = sn {
+                    sn
+                } else if let Some(snh) = snh {
+                    u128::from_str_radix(snh, 16)?
+                } else {
+                    0_u128
+                };
+                buf = sn.to_be_bytes();
```

Comment:
I haven't looked at the Python implementation but I'm guessing this is always exactly 8 octets/bytes right?

Reply:
Sorry 16

Reply:
Yes, raw is always 16 bytes, and u128.to_be_bytes always returns 16 bytes buffer.